### PR TITLE
Numpy 2 Support

### PR DESCRIPTION
**Pull request by AnnaBruenisholz**
Addresses #7477. Should work with both numpy 1 and 2 now. 
## Boost

Requires the patch https://github.com/boostorg/python/pull/432 
This patch has already been merged into master and hence should be included in the next release of boost.

## Pandas
pandas >= 2.2.2 (https://pandas.pydata.org/pandas-docs/stable/whatsnew/v2.2.2.html) required.

### CODE REVIEW

**Comment by greglandrum** [comment]
I have tested this on my linux box with numpy v2.0 installed from conda and a self-installed version of boost 1.85 with the patch mentioned above applied.
All tests pass and everything looks good.

File: Code/GraphMol/Descriptors/test3D.py
Comment (greglandrum) [inline Code/GraphMol/Descriptors/test3D.py:1]: wow. I didn't think we still had any of this old python2 files still in the repo.
These could actually be removed in some future cleanup PR

File: Code/GraphMol/DistGeomHelpers/Wrap/rdDistGeom.cpp [hunk 1]
```diff
@@ -78,6 +78,7 @@ namespace {
       if (nrows <= 0) {
         throw_value_error("The array has to have a nonzero size");
       }
+
```
Comment (greglandrum) [inline Code/GraphMol/DistGeomHelpers/Wrap/rdDistGeom.cpp:81]: This is just a formatting change and does not need to be in this PR.

**Review by greglandrum — CHANGES_REQUESTED** [review]
A very minor change suggested, otherwise this looks great.

Thanks!

**Review by greglandrum — APPROVED** [review]
LGTM

**Comment by greglandrum** [comment]
I am surprised (and VERY happy) that this didn't end up requiring a huge number of changes.

Thanks for the analysis and doing the work @AnnaBruenisholz !

**Comment by jakirkham** [comment]
Thank you both! 🙏

**Pull request merged**